lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -450,12 +596,15 @@ export class BrowserStorage extends CacheManager {
     getCachedRequest(state: string, browserCrypto: ICrypto): AuthorizationCodeRequest {
         try {
             // Get token request from cache and parse as TokenExchangeParameters.
-            const encodedTokenRequest = this.getItem(this.generateCacheKey(TemporaryCacheKeys.REQUEST_PARAMS), CacheSchemaType.TEMPORARY) as string;
+            const encodedTokenRequest = this.getTemporaryCache(TemporaryCacheKeys.REQUEST_PARAMS, true);
+
             const parsedRequest = JSON.parse(browserCrypto.base64Decode(encodedTokenRequest)) as AuthorizationCodeRequest;
             this.removeItem(this.generateCacheKey(TemporaryCacheKeys.REQUEST_PARAMS));
+
             // Get cached authority and use if no authority is cached with request.
             if (StringUtils.isEmpty(parsedRequest.authority)) {
-                const cachedAuthority: string = this.getItem(this.generateAuthorityKey(state), CacheSchemaType.TEMPORARY) as string;
+                const authorityCacheKey: string = this.generateAuthorityKey(state);
+                const cachedAuthority: string = this.getTemporaryCache(authorityCacheKey);
```

Comment:
Does this double wrap the key? Once with `generateAuthorityKey` and again with the key generation inside `getTemporaryCache`? Is that ok?

Reply:
getTemporaryCache takes a boolean which generates a cacheKey if `true`.